packages/core/src/tools/error.spec.ts

```diff
@@ -107,3 +188,38 @@ describe('createHandlingStack', () => {
   at userCallOne @ (.*)`)
   })
 })
+
+describe('flattenErrorCauses', () => {
+  it('should return undefined if no cause found', () => {
+    const error = new Error('foo') as ErrorWithCause
+    const errorCauses = flattenErrorCauses(error, ErrorSource.LOGGER)
+    expect(errorCauses).toEqual(undefined)
+  })
+
+  it('should stop recursive loop if cause is not of type Error', () => {
+    const error = new Error('foo') as ErrorWithCause
+    const nestedError = { biz: 'buz', cause: new Error('boo') }
+    // @ts-ignore: want to test we can handle non error cases
+    error.cause = nestedError
+
+    const errorCauses = flattenErrorCauses(error, ErrorSource.LOGGER)
+    expect(errorCauses?.length).toEqual(undefined)
+  })
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏

- more precise test description
- avoid ts-ignore annotation
- inline unneeded variable

```suggestion
  it('should return undefined if cause is not of type Error', () => {
    const error = new Error('foo') as ErrorWithCause
    error.cause = { biz: 'buz', cause: new Error('boo') } as unknown as Error

    const errorCauses = flattenErrorCauses(error, ErrorSource.LOGGER)
    expect(errorCauses?.length).toEqual(undefined)
  })
```